lib/instrumentation/fastify.js

```diff
@@ -40,6 +51,22 @@ const setupRouteHandler = (shim, fastify) => {
 
     routeOptions.handler = newRouteHandler
   })
+
+  shim.wrap(fastify, 'addHook', function addWrapHook(shim, fn) {
+    return function wrappedAddHook() {
+      const args = shim.argsToArray.apply(shim, arguments)
+      if (REQUEST_HOOKS.includes(args[0])) {
+        const middlewareFunction = args[1]
+        const newMiddlewareFunction = shim.recordMiddleware(
+          middlewareFunction,
+          buildMiddlewareSpecForMiddlewareFunction(shim, args[0])
```

Comment:
you may need to rebase but the 2nd are expects the segment name. so like this https://github.com/newrelic/node-newrelic/blob/main/lib/instrumentation/fastify.js#L52 but instead of hardcoding `onRequest` in that string you'd use `args[0] which is the lifecycle name